server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/dto/QuotaDTO.java

```diff
@@ -102,5 +107,13 @@ public String toString() {
             .toString();
     }
 
+    @JsonIgnore
+    public boolean isValid() {
+        return ImmutableList.of(count, size)
+            .stream()
+            .filter(quota -> quota.map(QuotaValue::asLong).orElse(Quotas.UNLIMITED) < Quotas.UNLIMITED)
```

Comment:
Something seems quite wrong here... and that's where the test `putQuotaShouldSetBothQuotaToInfinite` seems to fail. In the case of -1, if I understood correctly you apply an `Optional.empty()` to the `QuotaValue`.

However, `asLong` is defined like this in `QuotaCount` and `QuotaSize`:

```
return value.orElseThrow(IllegalStateException::new);
```

which throws the exception, in this case of `Optional.empty()`